lib/magma/server/update.rb

```diff
@@ -55,4 +60,119 @@ def load_revisions
+          |attribute_name| is_file_attribute(revision.model, attribute_name)
+        }.each do |attribute_name|
+          # For the remove operations, have to figure
+          #   out the "current" link name in Metis,
+          #   using the stored Magma value.
+          if !revision[attribute_name] || revision[attribute_name] == '::blank'
+            retrieval = Magma::Retrieval.new(
+              revision.model,
+              [revision.record_name.to_s],
+              [attribute_name],
+              restrict: !@user.can_see_restricted?(@project_name)
+            )
+
+            retrieval.records.each do |record|
+              # Only add a link to remove on Metis if a copy already exists.
+              # If the attribute hasn't been set before, we'll ignore this.
+              if record[attribute_name]
+                # This Metis route isn't active yet, so don't do anything
+                # link_revisions.push({
+                #   source: "metis://#{@project_name}/magma/#{record[attribute_name][:path]}",
+                #   dest: nil
+                # })
+              end
+            end
+          elsif revision[attribute_name].start_with? 'metis://'
+            copy_revisions.push({
+              source: revision[attribute_name],
+              dest: "metis://#{@project_name}/magma/#{revision.to_loader[attribute_name][:filename]}"
+            })
+          end
+        end
+      end
+    end
+
+    if copy_revisions.length > 0
+      execute_bulk_copy_on_metis(copy_revisions)
+    end
+  end
+
+  def execute_bulk_copy_on_metis(revisions)
+    host = Magma.instance.config(:storage).fetch(:host)
+
+    client = Etna::Client.new(
+      "https://#{host}",
+      @user.token)
+
+    bulk_copy_route = client.routes.find { |r| r[:name] == 'file_bulk_copy' }
+
+    return unless bulk_copy_route
+
+    # At some point, when Metis supports changing project names,
+    # this parameter should be the old file project name (metis_file_location_parts[2]))
+    # and the new project name in the HMAC headers should
+    # be @project_name
+    path = client.route_path(
+      bulk_copy_route,
+      project_name: @project_name)
+
+    bulk_copy_params = {
+      revisions: JSON.generate(revisions)
+    }
+
+    # Now populate the standard headers
+    hmac_params = {
+      method: 'POST',
+      host: host,
+      path: path,
+
+      expiration: (DateTime.now + 10).iso8601,
+      id: 'magma',
+      nonce: SecureRandom.hex,
+      headers: bulk_copy_params,
+    }
+
+    hmac = Etna::Hmac.new(Magma.instance, hmac_params)
+
+    # These get re-encoded into query string by Etna Client
+    hmac_params = Rack::Utils.parse_query(
+      hmac.url_params[:query]).map { |k,v| [k.to_sym, v] }.to_h
+
+    client.send(
+      'query_request',
+      Net::HTTP::Post,
+      hmac.url_params[:path],
+      hmac_params.merge(bulk_copy_params))
+  rescue Etna::Error => e
+    log(e.message)
+    @errors.concat([e.message])
```

Comment:
This, presumably, destroys `success?`, which then causes `load_revisions` not to run?